plugins/catalog/src/alpha.tsx

```diff
@@ -65,12 +81,191 @@ export const CatalogSearchResultListItemExtension =
+              <LazyComponent />
+            </React.Suspense>
+          </ExtensionBoundary>
+        ),
+      });
+    },
+  });
+}
+
+const CatalogEntityTagFilter = createCatalogFilterExtension({
+  id: 'entity.tag',
+  loader: async () => {
+    const { EntityTagPicker } = await import('@backstage/plugin-catalog-react');
+    return <EntityTagPicker />;
+  },
+});
+
+const CatalogEntityKindFilter = createCatalogFilterExtension({
+  id: 'entity.kind',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      initialFilter: z.string().default('component'),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityKindPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityKindPicker initialFilter={config.initialFilter} />;
+  },
+});
+
+const CatalogEntityTypeFilter = createCatalogFilterExtension({
+  id: 'entity.type',
+  loader: async () => {
+    const { EntityTypePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityTypePicker />;
+  },
+});
+
+const CatalogEntityOwnerFilter = createCatalogFilterExtension({
+  id: 'entity.mode',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      mode: z.enum(['owners-only', 'all']).optional(),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityOwnerPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityOwnerPicker mode={config.mode} />;
+  },
+});
+
+const CatalogEntityNamespaceFilter = createCatalogFilterExtension({
+  id: 'entity.namespace',
+  loader: async () => {
+    const { EntityNamespacePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityNamespacePicker />;
+  },
+});
+
+const CatalogEntityLifecycleFilter = createCatalogFilterExtension({
+  id: 'entity.lifecycle',
+  loader: async () => {
+    const { EntityLifecyclePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityLifecyclePicker />;
+  },
+});
+
+const CatalogEntityProcessingStatusFilter = createCatalogFilterExtension({
+  id: 'entity.processing.status',
+  loader: async () => {
+    const { EntityProcessingStatusPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityProcessingStatusPicker />;
+  },
+});
+
+const CatalogUserListFilter = createCatalogFilterExtension({
+  id: 'user.list',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      initialFilter: z.enum(['owned', 'starred', 'all']).default('owned'),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { UserListPicker } = await import('@backstage/plugin-catalog-react');
+    return <UserListPicker initialFilter={config.initialFilter} />;
+  },
+});
+
+const CatalogIndexPage = createPageExtension({
+  id: 'catalog',
+  defaultPath: '/catalog',
+  routeRef: rootRouteRef,
+  inputs: {
+    filters: createExtensionInput({
+      element: coreExtensionData.reactElement,
```

Comment:
This one is interesting. This lets people add any "component" (actually element) of any type here, even non-filters. We mentioned this in some meeting right? Not sure if we should worry about it or not.

This goes two ways. Even if we add some second discriminator data field, we still have the opposite problem that it gets possible to mount entity filters on generic reactElement inputs - where they will inevitably fail to render since there's no entity filter context around them.

Hm.

Should we add a

```ts
export const catalogExtensionData = {
  catalogFilter: createExtensionDataRef<{ element: JSX:Element }>('catalog.filter'),
};
```

in `@backstage/plugin-catalog-react`? Or is this just wasteful? @Rugvip

Reply:
@Rugvip mentioned possibly using a type similar to scaffolder fields, interested to hear your thoughts as well @freben:
https://github.com/backstage/backstage/pull/20355#discussion_r1348830910,

Reply:
alright, so we leave it as reactElement for now and possibly revisit later, if i'm reading that right?

Reply:
This is my understanding too, but let's double check wit @Rugvip 🙂

Reply:
Yep I'm honestly even a bit torn on migrating this at this point. I feel the system has come far enough that we'd be able to build a proper abstraction around catalog filters instead. Migrating to an API that mimics the existing react-based APIs might be counter productive in that we're fairly certain we'll want to remove it in the future and we don't really need it right now.